src/plugins/files/server/blob_storage_service/adapters/es/es.ts

```diff
@@ -169,8 +188,8 @@ export class ElasticsearchBlobStorageClient implements BlobStorageClient {
     return lastValueFrom(defer(processUpload).pipe(this.uploadSemaphore.acquire()));
   }
 
-  private getReadableContentStream(id: string, size?: number): ReadableContentStream {
-    return getReadableContentStream({
+  private getReadableContentStream(id: string, size?: number): () => ReadableContentStream {
```

Comment:
Maybe this wrapping into a function is not necessary, I've tried unwrapping it and tests still pass:

<img width="1445" alt="image" src="https://github.com/elastic/kibana/assets/82822460/607e27bc-251c-427b-a71e-935f66146679">

Having said that, maybe there is something else that requires it?

Reply:
You're right it does work without wrapping it in the current iteration.